x-pack/legacy/plugins/epm/server/lib/field.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// This should become a copy of https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/mapping/field.go#L39
+export interface Field {
+  name: string;
+  type: string;
+  required: boolean;
+  description: string;
+  fields: Field[];
+}
+
+type Fields = Field[];
+
+/**
+ * ProcessFields takes the given fields read from yaml and expands it.
+ * There are dotted fields in the field.yml like `foo.bar`. These should
+ * be stored as an object inside an object and is the main purpose of this
+ * preprocessing.
+ */
+export function ProcessFields(fields: Fields) {
```

Comment:
The convention in JS is only to use "PascalCase" for classes or thing which are invoked with `new`. TS convention seems to be to also use them for type/interface names.

In both, functions are named in `lowerCamelCase` style